x-pack/plugins/fleet/server/services/secrets.ts

```diff
@@ -237,6 +244,105 @@ export async function extractAndWriteSecrets(opts: {
   };
 }
 
+export async function extractAndWriteOutputSecrets(opts: {
+  output: NewOutput;
+  esClient: ElasticsearchClient;
+}): Promise<{ output: NewOutput; secretReferences: PolicySecretReference[] }> {
+  const { output, esClient } = opts;
+
+  const secretPaths = getOutputSecretPaths(output.type, output).filter(
+    (path) => typeof path.value === 'string'
+  );
+
+  if (secretPaths.length === 0) {
+    return { output, secretReferences: [] };
+  }
+
+  const secrets = await createSecrets({
+    esClient,
+    values: secretPaths.map(({ value }) => value as string),
+  });
+
+  const outputWithSecretRefs = JSON.parse(JSON.stringify(output));
+  secretPaths.forEach((secretPath, i) => {
+    set(outputWithSecretRefs, secretPath.path, { id: secrets[i].id });
+  });
+
+  return {
+    output: outputWithSecretRefs,
+    secretReferences: secrets.map(({ id }) => ({ id })),
+  };
+}
+
+function getOutputSecretPaths(
```

Comment:
It would be nice to document in a readme, where we have to make changes to add a new output secret e.g. `remote-elasticsearch` service token